### PR TITLE
SRIOV lane: skip flaky tests

### PR DESCRIPTION
**Pull request by ormergi**
Signed-off-by: Or Mergi <ormergi@redhat.com>

**What this PR does / why we need it**:
In order to enable sriov lane back and save CI resources  and developers time
skip flaky tests on sriov lane until their issues are resolved:
https://github.com/kubevirt/kubevirt/issues/3840
https://github.com/kubevirt/kubevirt/issues/3774
https://github.com/kubevirt/kubevirt/issues/3747
https://github.com/kubevirt/kubevirt/issues/4432

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by oshoval** [comment]
Please add the issue i opened https://github.com/kubevirt/kubevirt/issues/4432

**Review by oshoval — CHANGES_REQUESTED** [review]
Please add the issue i opened https://github.com/kubevirt/kubevirt/issues/4432

**Comment by EdDev** [comment]
> Please add the issue i opened #4432
> as the others would be solved by #4001

Add it where?

**Comment by oshoval** [comment]
> > Please add the issue i opened #4432
> > as the others would be solved by #4001
> 
> Add it where?

Either in the code, or in the PR description, or both
it simple to simulate and verify it will cause flakes after 4001 is merged, simply add delay before the ping,
info in the issue.
This is to the best of my knowledge,  i saw it happens locally when i worked on SRIOV (and had alternative fix for 4001 locally).
Else the skip in the code is not accurate, as there is another reason why the test would flake, even after 4001 is merged, closing the 3 mentioned issue.

**Comment by ormergi** [comment]
> > > Please add the issue i opened #4432
> > > as the others would be solved by #4001
> > 
> > 
> > Add it where?
> 
> Either in the code, or in the PR description, or both
> it simple to simulate and verify it will cause flakes after 4001 is merged, simply add delay before the ping,
> info in the issue.
> This is to the best of my knowledge, i saw it happens locally when i worked on SRIOV (and had alternative fix for 4001 locally).
> Else the skip in the code is not accurate, as there is another reason why the test would flake, even after 4001 is merged, closing the 3 mentioned issue.

I added it to the PR description

**Comment by oshoval** [comment]
> > > > Please add the issue i opened #4432
> > > > as the others would be solved by #4001
> > > 
> > > 
> > > Add it where?
> > 
> > 
> > Either in the code, or in the PR description, or both
> > it simple to simulate and verify it will cause flakes after 4001 is merged, simply add delay before the ping,
> > info in the issue.
> > This is to the best of my knowledge, i saw it happens locally when i worked on SRIOV (and had alternative fix for 4001 locally).
> > Else the skip in the code is not accurate, as there is another reason why the test would flake, even after 4001 is merged, closing the 3 mentioned issue.
> 
> I added it to the PR description

would you please add it to the code as well? 
if we trust the method i wrote there, its legitimate to add it there please as well, as its more informative

**Review by oshoval — CHANGES_REQUESTED** [review]
please add the skip to the tests,
you believe what i wrote in the issue is right if i understand correct.

**Comment by ormergi** [comment]
> > > > > Please add the issue i opened #4432
> > > > > as the others would be solved by #4001
> > > > 
> > > > 
> > > > Add it where?
> > > 
> > > 
> > > Either in the code, or in the PR description, or both
> > > it simple to simulate and verify it will cause flakes after 4001 is merged, simply add delay before the ping,
> > > info in the issue.
> > > This is to the best of my knowledge, i saw it happens locally when i worked on SRIOV (and had alternative fix for 4001 locally).
> > > Else the skip in the code is not accurate, as there is another reason why the test would flake, even after 4001 is merged, closing the 3 mentioned issue.
> > 
> > 
> > I added it to the PR description
> 
> would you please add it to the code as well?
> if we trust the method i wrote there, its legitimate to add it there please as well, as its more informative

I dont think we need to add another link to the skip.
The issues that I refer to are specific to each test as they were reported
Your issue is mentioned on the PR description as you requested, 
you can add a link to your issue from those that I mentioned since its 
describe the troubleshoot you did and possible solutions.

**Comment by oshoval** [comment]
> > > > > > Please add the issue i opened #4432
> > > > > > as the others would be solved by #4001
> > > > > 
> > > > > 
> > > > > Add it where?
> > > > 
> > > > 
> > > > Either in the code, or in the PR description, or both
> > > > it simple to simulate and verify it will cause flakes after 4001 is merged, simply add delay before the ping,
> > > > info in the issue.
> > > > This is to the best of my knowledge, i saw it happens locally when i worked on SRIOV (and had alternative fix for 4001 locally).
> > > > Else the skip in the code is not accurate, as there is another reason why the test would flake, even after 4001 is merged, closing the 3 mentioned issue.
> > > 
> > > 
> > > I added it to the PR description
> > 
> > 
> > would you please add it to the code as well?
> > if we trust the method i wrote there, its legitimate to add it there please as well, as its more informative
> 
> I dont think we need to add another link to the skip.
> The issues that I refer to are specific to each test as they were reported
> Your issue is mentioned on the PR description as you requested,
> you can add a link to your issue from those that I mentioned since its
> describe the troubleshoot you did and possible solutions.

sorry i dont accept it, if you know this issue is real, and once 4001 would be merged, people might remove the skip and then it would cause CI to be flaky, i would ask please that the code would reflect it.

**Comment by ormergi** [comment]
> sorry i dont accept it, if you know this issue is real, and once 4001 would be merged, people might remove the skip and then it would cause CI to be flaky, i would ask please that the code would reflect it.
> its a different problem as we spoke, so i dont need to link and cause misinformation.

I dont know what your taking about,  #3840 will be resolved by 4001
#3774 and #3747 describing flaky tests and thats it.
#4432 the one you open is suggesting a solution to #3774 and #3747

**Comment by oshoval** [comment]
> > sorry i dont accept it, if you know this issue is real, and once 4001 would be merged, people might remove the skip and then it would cause CI to be flaky, i would ask please that the code would reflect it.
> > its a different problem as we spoke, so i dont need to link and cause misinformation.
> 
> I dont know what your taking about, #3840 will be resolved by 4001
> #3774 and #3747 describing flaky tests and thats it.
> #4432 the one you open is suggesting a solution to #3774 and #3747



> > sorry i dont accept it, if you know this issue is real, and once 4001 would be merged, people might remove the skip and then it would cause CI to be flaky, i would ask please that the code would reflect it.
> > its a different problem as we spoke, so i dont need to link and cause misinformation.
> 
> I dont know what your taking about, #3840 will be resolved by 4001
> #3774 and #3747 describing flaky tests and thats it.
> #4432 the one you open is suggesting a solution to #3774 and #3747

the timeout is about the bug 4001 fix (3774 , 3747 topic)
if those are passing but the phenomena 4432 say occur the error will be ping failed.
its not the same bug

**Comment by ormergi** [comment]
> the timeout is about the bug 4001 fix (3774 , 3747 topic)
> if those are passing but the phenomena 4432 say occur the error will be ping failed.
> its not the same bug

There is a timeout because of what #3840 describes
#3774 and #3747 are reported by CI officer, they will not debug it they just reported what thy saw on the job,
the solution for those tests will be the same for both.
Those issues wont be resolved until there is a PR to fix them so you dont need to worry 
that if there is no timeout it is solved.

**Comment by oshoval** [comment]
> > the timeout is about the bug 4001 fix (3774 , 3747 topic)
> > if those are passing but the phenomena 4432 say occur the error will be ping failed.
> > its not the same bug
> 
> There is a timeout because of what #3840 describes
> #3774 and #3747 are reported by CI officer, they will not debug it they just reported what thy saw on the job,
> the solution for those tests will be the same for both.
> Those issues wont be resolved until there is a PR to fix them so you dont need to worry
> that if there is no timeout it is solved.

Sorry, i dont accept it, those are two different bugs, 
4432 solves bug that wasnt known publicly before i mentioned it, it was assumed the interface selection would solve it.
Again i wrote the phenomena even are different. so there shouldn't be a cross link.

**Comment by ormergi** [comment]
> > > the timeout is about the bug 4001 fix (3774 , 3747 topic)
> > > if those are passing but the phenomena 4432 say occur the error will be ping failed.
> > > its not the same bug
> > 
> > 
> > There is a timeout because of what #3840 describes
> > #3774 and #3747 are reported by CI officer, they will not debug it they just reported what thy saw on the job,
> > the solution for those tests will be the same for both.
> > Those issues wont be resolved until there is a PR to fix them so you dont need to worry
> > that if there is no timeout it is solved.
> 
> Sorry, i dont accept it, those are two different bugs,
> 4432 solves bug that wasnt known publicly before i mentioned it, it was assumed the interface selection would solve it.
> Again i wrote the phenomena even are different. so there shouldn't be a cross link.

The fact that there is a comment that says it might solve it dosnt mean it is assumed that the interface selection would solve it.
This is just for skipping tests that will be resolved next week, and the issues that mentioned in the code reported they are flaky thats all.
It dosesnt explain what exactly is the problem just what the logs show.. take a looke at  https://github.com/kubevirt/kubevirt/issues/3774 it is not even mentioned what is the problem just that the test was failing

Please stop blocking this PR we want to continue with fixing SRIOV lane and get initial feedback about its health
lets dont waste more energy on this :)

**Comment by oshoval** [comment]
> > > > the timeout is about the bug 4001 fix (3774 , 3747 topic)
> > > > if those are passing but the phenomena 4432 say occur the error will be ping failed.
> > > > its not the same bug
> > > 
> > > 
> > > There is a timeout because of what #3840 describes
> > > #3774 and #3747 are reported by CI officer, they will not debug it they just reported what thy saw on the job,
> > > the solution for those tests will be the same for both.
> > > Those issues wont be resolved until there is a PR to fix them so you dont need to worry
> > > that if there is no timeout it is solved.
> > 
> > 
> > Sorry, i dont accept it, those are two different bugs,
> > 4432 solves bug that wasnt known publicly before i mentioned it, it was assumed the interface selection would solve it.
> > Again i wrote the phenomena even are different. so there shouldn't be a cross link.
> 
> The fact that there is a comment that says it might solve it dosnt mean it is assumed that the interface selection would solve it.
> This is just for skipping tests that will be resolved next week, and the issues that mentioned in the code reported they are flaky thats all.
> It dosesnt explain what exactly is the problem just what the logs show.. take a looke at #3774 it is not even mentioned what is the problem just that the test was failing
> 
> Please stop blocking this PR we want to continue with fixing SRIOV lane and get initial feedback about its health
> lets dont waste more energy on this :)

Sorry, see, in one of the first 3 , that Alona said it cant find eth1, it sumup that the 15 seconds timeout is one bug
and 4432 is other in those 2 tests, once the interface is fine.
I am not blocking, i dont give my lgtm since its not lgtm.

**Comment by ormergi** [comment]
> As we discussed you agreed that even making 1 interface wont solve the ping, i expect the PR code to mention the issue.
> We have here an issue, standalone one, that we agree is mandatory, it must be a skip line.

I didnt say your issue dont describe the right problem its describe the troubleshoot you dod.
I mention other issues in the code because they are a report for flaky test, we dont need to get into details here.
You asked for the issue to be mentioned in the code or on the PR description, so I added to the PR description
why you keep on blocking this?
I didnt ask your lgtm, it was lgtmed already you removed it with your comment

**Comment by oshoval** [comment]
> > As we discussed you agreed that even making 1 interface wont solve the ping, i expect the PR code to mention the issue.
> > We have here an issue, standalone one, that we agree is mandatory, it must be a skip line.
> 
> I didnt say your issue dont describe the right problem its describe the troubleshoot you dod.
> I mention other issues in the code because they are a report for flaky test, we dont need to get into details here.
> You asked for the issue to be mentioned in the code or on the PR description, so I added to the PR description
> why you keep on blocking this?
> I didnt ask your lgtm, it was lgtmed already you removed it with your comment

I ask that it will be in the code as well,
reasons are allover above,
I have the right to request changes.

**Comment by ormergi** [comment]
> I ask that it will be in the code as well,
https://github.com/kubevirt/kubevirt/pull/4438#issuecomment-718000736

> reasons are allover above, we do need to get into details, as the other issues will be auto closed once 4001 is merged,
> and without the additional Skip, the code wont present it as it should and does on every test with a Skip.
> I have the right to request changes.
> I worked hard and found this, and reported it, so the CI wont suffer flakes, before it hit the CI.
> All the info, debugging, reason, and solution are in the issue.
> We are in the last mile, lets do it right please.

Those issues wont close automatically, why you think they will?
Sure you have the right to ask for changes, but why blocking an already lgtmed PR because you issue is mentioned in the PR and not in the code?
Like you said we are on the last mile after weeks on working to fix sriov lane, 
if you dont want to lgtm this its ok, dont block it.

**Comment by oshoval** [comment]
> > I ask that it will be in the code as well,
> > [#4438 (comment)](https://github.com/kubevirt/kubevirt/pull/4438#issuecomment-718000736)
> 
> > reasons are allover above, we do need to get into details, as the other issues will be auto closed once 4001 is merged,
> > and without the additional Skip, the code wont present it as it should and does on every test with a Skip.
> > I have the right to request changes.
> > I worked hard and found this, and reported it, so the CI wont suffer flakes, before it hit the CI.
> > All the info, debugging, reason, and solution are in the issue.
> > We are in the last mile, lets do it right please.
> 
> Those issues wont close automatically, why you think they will?
> Sure you have the right to ask for changes, but why blocking an already lgtmed PR because you issue is mentioned in the PR and not in the code?
> Like you said we are on the last mile after weeks on working to fix sriov lane,
> if you dont want to lgtm this its ok, dont block it.

Because 3 hours ago there were mentioned on #4001 fix list,
and on any case, the issue i asked to add to the code is describing an error the others doesn't describe, so it must be included,
moreover we even saw on CI itself on one of the runs this issue

File: tests/vmi_multus_test.go
Comment (oshoval) [inline tests/vmi_multus_test.go:871]: since 4001 points now on 3840, this should point on 3840 as well please.
(and also on 4432)

see please
https://github.com/kubevirt/kubevirt/pull/4438#issuecomment-718091502


same for line 876

File: tests/vmi_multus_test.go
Comment (oshoval) [inline tests/vmi_multus_test.go:876]: space at the end,
also see please
https://github.com/kubevirt/kubevirt/pull/4438/files#r514057626

**Comment by ormergi** [comment]
> Because 3 hours ago there were mentioned on #4001 fix list,
> and on any case, even since its edited 3 hours ago and it wont close them 2 of them,
> the issue i asked to add to the code is describing an error the others doesn't describe, so it must be included,
> moreover we even saw on CI itself on one of the runs this issue (when ran manually),
> and those other two would be left orphan until they auto closed, so this PR still point them and not sure it points right,
> it would be better to point to the 2 issues only, the one 4001 says it fixes, and 4432, in the code itself,
> instead of pointing to some other issues just because they have the test_id.

For those tests case, we see the issue that described in 3840 affect all 3 tests but solves the problem for one of them.
3774 and 3747 issues are just a reports for flaky tests and this is enough.
I can give you 20 more jobs that show that the ping test fails due to ping failure (destination unreachable),
once the ip configuration will fixes we will close them.
The issue you opened add more detail that were not described erlier you right, this is why I mention it in the PR
and the other reports for flakiness in the code.
I get it you dont accept it, you dont have to.

**Comment by oshoval** [comment]
> > Because 3 hours ago there were mentioned on #4001 fix list,
> > and on any case, even since its edited 3 hours ago and it wont close them 2 of them,
> > the issue i asked to add to the code is describing an error the others doesn't describe, so it must be included,
> > moreover we even saw on CI itself on one of the runs this issue (when ran manually),
> > and those other two would be left orphan until they auto closed, so this PR still point them and not sure it points right,
> > it would be better to point to the 2 issues only, the one 4001 says it fixes, and 4432, in the code itself,
> > instead of pointing to some other issues just because they have the test_id.
> 
> For those tests case, we see the issue that described in 3840 affect all 3 tests but solves the problem for one of them.
> 3774 and 3747 issues are just a reports for flaky tests and this is enough.
> I can give you 20 more jobs that show that the ping test fails due to ping failure (destination unreachable),
> once the ip configuration will fixes we will close them.
> The issue you opened add more detail that were not described erlier you right, this is why I mention it in the PR
> and the other reports for flakiness in the code.
> I get it you dont accept it, you dont have to.

https://github.com/kubevirt/kubevirt/pull/4438#discussion_r514057626
for more info
its wrong to point to issues just because they have the test_id,
moreoever all the 3 issues say timeout of 15 seconds because the eth1 wasnt found,
and when this stage passed , 4432 shows another error, which 4432 express exactly a preliminary fix that must be done.

**Comment by alonSadan** [comment]
copied from (https://github.com/kubevirt/kubevirt/pull/4001#issuecomment-718470486):

> This PR could fix #3840, I almost sure it wont fix #3747 and #3774
Could you please update the PR description?

@ormergi please take a look here and here at what @oshoval said.

I think this PR will fix those issues, but not #4432. even though they affect the same test , I don't think this is the same issue:

the first two, when failing states:

Sent: "ip addr add 192.168.1.1/24 dev eth1\n"
Match for RE: "#" found: ["#"] Buffer: ip addr add 192.168.1.1/24 dev eth1
Cannot find device "eth1"

and the last one states:

Sent: "ping 192.168.1.2 -c 1 -w 5\n"
Match for RE: "(\\$ |\\# )" found: ["# " "# "] Buffer: ping 192.168.1.2 -c 1 -w 5
PING 192.168.1.2 (192.168.1.2) 56(84) bytes of data.

--- 192.168.1.2 ping statistics ---
5 packets transmitted, 0 received, 100% packet loss, time 4078ms

[root@testvmignfs6fkv244dtqd4r5snbbprgkk6rzrrsqgw8lwh4gws4zr4 fedora]# 
I think the the first one is caused by the mixup with the names of the interface, which hopefully this PR will solve.

the last one is caused by the disappear of the IP from the vm.
So I think we should close those issues once this PR is merged, and then tackle #4432.
what do you think?

**Comment by ormergi** [comment]
> [#4438 (comment)](https://github.com/kubevirt/kubevirt/pull/4438#discussion_r514057626)
> for more info
> its wrong to point to issues just because they have the test_id,
> moreoever all the 3 issues say timeout of 15 seconds because the eth1 wasnt found,
> and when this stage passed , 4432 shows another error, which 4432 express exactly a preliminary fix that must be done.





> > > Because 3 hours ago there were mentioned on #4001 fix list,
> > > and on any case, even since its edited 3 hours ago and it wont close them 2 of them,
> > > the issue i asked to add to the code is describing an error the others doesn't describe, so it must be included,
> > > moreover we even saw on CI itself on one of the runs this issue (when ran manually),
> > > and those other two would be left orphan until they auto closed, so this PR still point them and not sure it points right,
> > > it would be better to point to the 2 issues only, the one 4001 says it fixes, and 4432, in the code itself,
> > > instead of pointing to some other issues just because they have the test_id.
> > 
> > 
> > For those tests case, we see the issue that described in 3840 affect all 3 tests but solves the problem for one of them.
> > 3774 and 3747 issues are just a reports for flaky tests and this is enough.
> > I can give you 20 more jobs that show that the ping test fails due to ping failure (destination unreachable),
> > once the ip configuration will fixes we will close them.
> > The issue you opened add more detail that were not described erlier you right, this is why I mention it in the PR
> > and the other reports for flakiness in the code.
> > I get it you dont accept it, you dont have to.
> 
> [#4438 (comment)](https://github.com/kubevirt/kubevirt/pull/4438#discussion_r514057626)
> for more info
> its wrong to point to issues just because they have the test_id,
> moreoever all the 3 issues say timeout of 15 seconds because the eth1 wasnt found,
> and when this stage passed , 4432 shows another error, which 4432 express exactly a preliminary fix that must be done.

Keep this discussion here.
You are mixing, the issues 3774 and 3747  report the flakiness they also refer to the last time the problem was seen in the log, it just a matter of "luck" what problem would affect those tests..
The tests that those issues refer to affected by two different issues.
I prefer to add a link to the flakyiness report issue

**Comment by oshoval** [comment]
> > [#4438 (comment)](https://github.com/kubevirt/kubevirt/pull/4438#discussion_r514057626)
> > for more info
> > its wrong to point to issues just because they have the test_id,
> > moreoever all the 3 issues say timeout of 15 seconds because the eth1 wasnt found,
> > and when this stage passed , 4432 shows another error, which 4432 express exactly a preliminary fix that must be done.
> 
> Keep this discussion here.
> You are mixing, the issues 3774 and 3747 report the flakiness they also refer to the last time the problem was seen in the log, it just a matter of "luck" what problem would affect those tests..
> The tests that those issues refer to affected by two different issues.
> I prefer to add a link to the flakyiness report issue

one of the lanes you manually ran shows on prow itself the ping failure, and then i explained its another problem that 4001 wont solve, and presented 4432, please attach that link as well.

If the eth1 isnt found, it wont reach the chance to try and ping, so those are two different issues that affect those two tests that mentioned on 4432.
There is a chain of flow here, with some deterministic stuff that we know, and the deterministic stuff should be referenced in the Skip until its fixed.
At the moment, you even point to two issues that dont have a PR pointing to them as fixing them,
as i wrote above its an issue as well.

**Comment by oshoval** [comment]
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4424/pull-kubevirt-e2e-kind-1.17-sriov/1320997424440807424

see ping failure here, CI shouldn't suffer from it, when we knowing a smoking gun.

**Comment by ormergi** [comment]
> > > [#4438 (comment)](https://github.com/kubevirt/kubevirt/pull/4438#discussion_r514057626)
> > > for more info
> > > its wrong to point to issues just because they have the test_id,
> > > moreoever all the 3 issues say timeout of 15 seconds because the eth1 wasnt found,
> > > and when this stage passed , 4432 shows another error, which 4432 express exactly a preliminary fix that must be done.
> > 
> > 
> > Keep this discussion here.
> > You are mixing, the issues 3774 and 3747 report the flakiness they also refer to the last time the problem was seen in the log, it just a matter of "luck" what problem would affect those tests..
> > The tests that those issues refer to affected by two different issues.
> > I prefer to add a link to the flakyiness report issue
> 
> one of the lanes you manually ran shows on prow itself the ping failure, and then i explained its another problem that 4001 wont solve, and presented 4432, please attach that link as well.
> 
> If the eth1 isnt found, it wont reach the chance to try and ping, so those are two different issues that affect those two tests that mentioned on 4432.
> There is a chain of flow here, with some deterministic stuff that we know, and the deterministic stuff should be referenced in the Skip until its fixed.
> At the moment, you even point to two issues that dont have a PR pointing to them as fixing them,
> as i wrote above its an issue as well.

We have been over it, let see what other people think.
So to sum it up, there are two issues that report falkyness for the ping connectivity tests that I refer to in the code and in the PR description.
@oshoval opend an issue that  describe the problem that cause the flakiness for those tests, this issue is mentioned on the PR description, but Or want it to be mentioned in the code instead of the falkyness report issues.

Please share your thoughts
@qinqon @phoracek @EdDev

**Comment by oshoval** [comment]
> > > > [#4438 (comment)](https://github.com/kubevirt/kubevirt/pull/4438#discussion_r514057626)
> > > > for more info
> > > > its wrong to point to issues just because they have the test_id,
> > > > moreoever all the 3 issues say timeout of 15 seconds because the eth1 wasnt found,
> > > > and when this stage passed , 4432 shows another error, which 4432 express exactly a preliminary fix that must be done.
> > > 
> > > 
> > > Keep this discussion here.
> > > You are mixing, the issues 3774 and 3747 report the flakiness they also refer to the last time the problem was seen in the log, it just a matter of "luck" what problem would affect those tests..
> > > The tests that those issues refer to affected by two different issues.
> > > I prefer to add a link to the flakyiness report issue
> > 
> > 
> > one of the lanes you manually ran shows on prow itself the ping failure, and then i explained its another problem that 4001 wont solve, and presented 4432, please attach that link as well.
> > If the eth1 isnt found, it wont reach the chance to try and ping, so those are two different issues that affect those two tests that mentioned on 4432.
> > There is a chain of flow here, with some deterministic stuff that we know, and the deterministic stuff should be referenced in the Skip until its fixed.
> > At the moment, you even point to two issues that dont have a PR pointing to them as fixing them,
> > as i wrote above its an issue as well.
> 
> We have been over it, let see what other people think.
> So to sum it up, there are two issues that report falkyness for the ping connectivity tests that I refer to in the code and in the PR description.
> @oshoval opend an issue that describe the problem that cause the flakiness for those tests, this issue is mentioned on the PR description, but Or want it to be mentioned in the code instead of the falkyness report issues.
> 
> Please share your thoughts
> @qinqon @phoracek @EdDev

not instead, in addition, as its two problems 
the mix of the order and the ip that goes away.

**Comment by ormergi** [comment]
> > We have been over it, let see what other people think.
> > So to sum it up, there are two issues that report falkyness for the ping connectivity tests that I refer to in the code and in the PR description.
> > @oshoval opend an issue that describe the problem that cause the flakiness for those tests, this issue is mentioned on the PR description, but Or want it to be mentioned in the code instead of the falkyness report issues.
> > Please share your thoughts
> > @qinqon @phoracek @EdDev
> 
> not instead, in addition, as its two problems:
> 
> 1. the mix of the order
> 2. the ip that goes away. (i simulated exact phenomenon locally, issue have all info + possible solutions), see also link at the end here please.
> 
> Moreover see please the comments that you point to issues without a PR that fix them, all the 3 should point to what we know which is 3840 which is fixed by 4001
> 
> see ping error here
> https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4424/pull-kubevirt-e2e-kind-1.17-sriov/1320997424440807424

I am aware for all the problems and the root cuases for those tests problems.
The issues I mention in the code are a user reports for flaky tests, any issue that mention this problem is kind of a duplicate that should be referenced on the report issue.
It dosent matter what is the reason you see in the logs on the flaky report issue, what matter is that its report the a test is flaky.
once the tests flakyness is resolved the report issues will be closed and all other issues that refer to that flaky test.
This is why #3774 and #3747 issues are mentioned in the code, they are the report for flaky test by a user.

**Review by EdDev — APPROVED** [review]
The current referenced issues look good to me.
They point to users who detected the overall problem of the test failing. The fact that there are multiple issues that causes the problem is an inner detail. Referencing from those issues to the individual problems investigated is enough.

**Comment by EdDev** [comment]
> /hold please answer Or Shoval's comments.

The ref to the issues have been addressed. There is no information lost and the skip is managed from the user tickets that  must be resolved before closed. Details on the exact issues detected, are ref through other issues (which those issue point to and this PR mentions as well).

I think we are good.

/hold cancel

File: tests/vmi_multus_test.go
Comment (ormergi) [reply tests/vmi_multus_test.go:871]: #3774 is the first issue that opened by the user to report flakyness
any other issue that adding more to that is referenced in that issue.
As we discussed it in the main PR page lets continue there, 
I am resolving this comment

File: tests/vmi_multus_test.go
Comment (ormergi) [reply tests/vmi_multus_test.go]: Fixed spaces
#3774 is the first issue that opened by the user to report flakyness
any other issue that adding more to that is referenced in that issue.
As we discussed it in the main PR page lets continue there, 
I am resolving this comment

**Comment by ormergi** [comment]
Changes:
- Removed redundant spaces
- As for the ref issues see upper discussion

**Review by EdDev — APPROVED** [review]

**Comment by ormergi** [comment]
/retest

Jobs were stuck due to SSL cert rotation on CI cluster

**Comment by ormergi** [comment]
I removed the skip form the test #4001 fixes

**Comment by ormergi** [comment]
I removed the skip form the test #4001 fixes
@qinqon @phoracek

**Comment by oshoval** [comment]
> I removed the skip form the test #4001 fixes
> @qinqon @phoracek

imo the accurate is now #4432

**Comment by EdDev** [comment]
> I removed the skip form the test #4001 fixes
> @qinqon @phoracek

Why did you removed that skip? I lost you there.
All tests have the same 2 problems, we still have the issue with nm vs static ip setting using the ip tool.

**Comment by EdDev** [comment]
> IMO the accurate issue that we should reference is now #4432
> as this is actually exactly why we are skipping (referring to the first post on each issue)
> the current refereed issues has 4432 on its discussion (on the bottom of it, mixed with other details that are no longer the skip reason), i know, but still its less accurate from my perspective,
> as one would need to read the whole thread once we have an issue which directly have all the info.
> 
> As i told it above, i am telling it again just because the state of the PR is changed, the reviewers will decide.

Repeating the same argument does not make it stronger or more convincing.
It is fine that you disagree, but I see no point in going over this in a loop.
My last comment still stands, the ref issue Is about the test not passing, and it is general and not specific.

**Comment by oshoval** [comment]
> > I removed the skip form the test #4001 fixes
> > @qinqon @phoracek
> 
> Why did you removed that skip? I lost you there.
> All tests have the same 2 problems, we still have the issue with nm vs static ip setting using the ip tool.

I am remember right (according the issues above, the 3rd test was 3985 which doesnt set ip, so it has just the issue 4001) have,
since it use hardcoded "eth1"

**Comment by oshoval** [comment]
> > IMO the accurate issue that we should reference is now #4432
> > as this is actually exactly why we are skipping (referring to the first post on each issue)
> > the current refereed issues has 4432 on its discussion (on the bottom of it, mixed with other details that are no longer the skip reason), i know, but still its less accurate from my perspective,
> > as one would need to read the whole thread once we have an issue which directly have all the info.
> > As i told it above, i am telling it again just because the state of the PR is changed, the reviewers will decide.
> 
> Repeating the same argument does not make it stronger or more convincing.
> It is fine that you disagree, but I see no point in going over this in a loop.
> My last comment still stands, the ref issue Is about the test not passing, and it is general and not specific.

I didnt change lgtm status, i answered on the new PR states, which also say mentioned what you said, and explained what i think about it, as i didnt do it before on your opinio, ending with let the other reviewers decide.

Thanks

**Comment by oshoval** [comment]
This ongoing potential confusion wouldn't happen if we used Skip cascading as i suggested.

**Comment by ormergi** [comment]
> > I removed the skip form the test #4001 fixes
> > @qinqon @phoracek
> 
> IMO the accurate issue that we should reference is now #4432
> as this is actually exactly why we are skipping (referring to the first post on each issue)
> the current refereed issues has 4432 on its discussion (on the bottom of it, mixed with other details that are no longer the skip reason), i know, but still its less accurate from my perspective,
> as one would need to read the whole thread once we have an issue which directly have all the info.
> 
> As i told it above, i am telling it again just because the state of the PR is changed, the reviewers will decide.


> > I removed the skip form the test #4001 fixes
> > @qinqon @phoracek
> 
> IMO the accurate issue that we should reference is now #4432
> as this is actually exactly why we are skipping (referring to the first post on each issue)
> the current refereed issues has 4432 on its discussion (on the bottom of it, mixed with other details that are no longer the skip reason), i know, but still its less accurate from my perspective,
> as one would need to read the whole thread once we have an issue which directly have all the info.
> 
> As i told it above, i am telling it again just because the state of the PR is changed, the reviewers will decide.


> > I removed the skip form the test #4001 fixes
> > @qinqon @phoracek
> 
> IMO the accurate issue that we should reference is now #4432
> as this is actually exactly why we are skipping (referring to the first post on each issue)
> the current refereed issues has 4432 on its discussion (on the bottom of it, mixed with other details that are no longer the skip reason), i know, but still its less accurate from my perspective,
> as one would need to read the whole thread once we have an issue which directly have all the info.
> 
> As i told it above, i am telling it again just because the state of the PR is changed, the reviewers will decide.


> > I removed the skip form the test #4001 fixes
> > @qinqon @phoracek
> 
> Why did you removed that skip? I lost you there.
> All tests have the same 2 problems, we still have the issue with nm vs static ip setting using the ip tool.

The test that #4001 fixes configures custom mac address and check that it is there, 
we do not make any change to the interface (using `ip` command).

**Review by EdDev — APPROVED** [review]

**Pull request merged**
